services/provider/server/server.go

```diff
@@ -610,358 +608,172 @@ func decodeAndValidateTicket(ticket string, pubKey *rsa.PublicKey) (*services.On
 }
 
 // FulfillStorageClaim RPC call to create the StorageClaim CR on
-// provider cluster.
+// provider cluster is no longer supported since 4.19 so giving empty response.
 func (s *OCSProviderServer) FulfillStorageClaim(ctx context.Context, req *pb.FulfillStorageClaimRequest) (*pb.FulfillStorageClaimResponse, error) {
-	// Get storage consumer resource using UUID
-	consumerObj, err := s.consumerManager.Get(ctx, req.StorageConsumerUUID)
-	if err != nil {
-		return nil, status.Error(codes.Internal, err.Error())
-	}
-
-	klog.Infof("Found StorageConsumer %q (%q)", consumerObj.Name, req.StorageConsumerUUID)
-
-	var storageType string
-	switch req.StorageType {
-	case pb.FulfillStorageClaimRequest_BLOCK:
-		storageType = "block"
-	case pb.FulfillStorageClaimRequest_SHAREDFILE:
-		storageType = "sharedfile"
-	default:
-		return nil, status.Errorf(codes.InvalidArgument, "encountered an unknown storage type, %s", storageType)
-	}
-
-	err = s.storageRequestManager.Create(ctx, consumerObj, req.StorageClaimName, storageType, req.EncryptionMethod, req.StorageProfile)
-	if err != nil {
-		errMsg := fmt.Sprintf("failed to fulfill storage class claim for %q. %v", req.StorageConsumerUUID, err)
-		klog.Error(errMsg)
-		if kerrors.IsAlreadyExists(err) {
-			return nil, status.Error(codes.AlreadyExists, errMsg)
-		}
-		return nil, status.Error(codes.Internal, errMsg)
-	}
-
-	return &pb.FulfillStorageClaimResponse{}, nil
+	return nil, nil
 }
 
 // RevokeStorageClaim RPC call to delete the StorageClaim CR on
-// provider cluster.
+// provider cluster is no longer supported since 4.19 so giving failure.
 func (s *OCSProviderServer) RevokeStorageClaim(ctx context.Context, req *pb.RevokeStorageClaimRequest) (*pb.RevokeStorageClaimResponse, error) {
-	err := s.storageRequestManager.Delete(ctx, req.StorageConsumerUUID, req.StorageClaimName)
-	if err != nil {
-		errMsg := fmt.Sprintf("failed to revoke storage class claim %q for %q. %v", req.StorageClaimName, req.StorageConsumerUUID, err)
-		klog.Error(errMsg)
-		return nil, status.Error(codes.Internal, errMsg)
-	}
-
-	return &pb.RevokeStorageClaimResponse{}, nil
-}
-
-func storageClaimCephCsiSecretName(secretType, suffix string) string {
-	return fmt.Sprintf("ceph-client-%s-%s", secretType, suffix)
+	errMsg := fmt.Sprintf("failed to revoke storage class claim %q for %q.", req.StorageClaimName, req.StorageConsumerUUID)
+	klog.Error(errMsg)
+	return nil, status.Error(codes.Internal, errMsg)
```

Comment:
you are not actually failing to revoke but not proceeding with revoke which basically is an unsupported operation in 4.19, pls make the status code & message reflect the same.